psp_notes.md

```diff
@@ -0,0 +1,86 @@
+chainId (see authorizers in config.json master branch)
+
+This step means calling AllocatorWith\*.airnodeToSlotIndexToSlot(airnodeAddress). This will return a mappaing of index
+to Slot object where each object represents a susbscription.
+
+2. The Airnode fetches the subscription details from the chain if they have been stored (by calling
+   `storeSubscription()`), or the subscription could be hardcoded at the node configuration. It also fetches the
+   referenced template. It verifies the integrity of the template and the subscription details by comparing the hash of
+   the details to the ID.
+
+Similarly to call-api.ts in Airkeeper, subscription details will need to be hardcoded in the config or fetched from the
+chain by calling AirnodeProtocol.subscriptions(subscriptionId). Same for templates.
+
+3. The Airnode checks `AirnodeProtocol.sponsorToSubscriptionIdToPspSponsorshipStatus()` to verify the `sponsor`s
+   specified in the subscription details
+
+In previous RRP protocol v0 this mapping was used onchain to verify sponsorship status while making a request but now
+this check will only be performed off-chain, correct? There is a mention of this in the
+[DapiServer.registerBeaconUpdateSubscription()](https://github.com/api3dao/airnode/blob/v1-protocol/packages/airnode-protocol-v1/contracts/dapis/DapiServer.sol#L250-L252)
+function comments but no use of the mapping on-chain.
+
+4. The Airnode checks its Authorizers to verify that the `requester`s specified in the subscription details are
+   authorized
+
+Subscription details will set the `requester` to the DapiServer contract address.
+
+It sounds like we could keep the 2 config files approach currently being implemented in Airkeeper (but tbh I would
+really like to just update current config.json with all new fields required by PSP).
+
+See for reference:
+https://github.com/api3dao/airnode/blob/v1-protocol/packages/airnode-node/src/evm/authorization/authorization-fetching.ts#L45
+
+5. The Airnode makes the API call specified by the template and the additional parameters
+
+Probably current [call-api-ts](https://github.com/api3dao/airkeeper/blob/main/src/call-api.ts) in Airkeeper can be
+re-used for this step.
+
+6. Using the response from the API, the Airnode checks if the condition specified in the `conditions` field of the
+   subscription is met (in most cases, by making a static call to a function)
+
+I'm confused about
+[this](https://github.com/api3dao/airnode/blob/v1-protocol/packages/airnode-protocol-v1/contracts/dapis/DapiServer.sol#L20-L23)
+step because this comment says that it is not needed for beacon updates.
+
+If I still need to check the condition in Airkeeper then I guess I need to get the percentage value used when calling
+DapiServer.registerBeaconUpdateSubscription() and lastly call DapiServer.conditionPspBeaconUpdate(). The percentage
+value should be in the config but also in the subscription details from step 2.
```

Comment:
The percentage value should only be in the subscription details, it shouldn't be separately specified in the config file

Here's an example of how you would do that
https://github.com/api3dao/airnode/blob/v1-protocol/packages/airnode-protocol-v1/test/dapis/DapiServer.sol.js#L136
but I think we should do
```js
beaconUpdateSubscriptionConditions = hre.ethers.utils.defaultAbiCoder.encode(
    ['bytes32', 'bytes32', 'bytes32', 'bytes32', 'bytes'],
    [
      hre.ethers.utils.formatBytes32String('1bB'),
      hre.ethers.utils.formatBytes32String('_conditionFunctionId'),
      hre.ethers.utils.defaultAbiCoder.encode(
        ['bytes4'],
        [dapiServer.interface.getSighash('conditionPspBeaconUpdate')]
      ),
      hre.ethers.utils.formatBytes32String('_conditionParameters'),
      beaconUpdateSubscriptionConditionParameters,
    ]
  );
```
and the lack of `_conditionChainId` means you should use chain the requester contract is on and the lack of `_conditionAddress` means you should use the requester contract address (so that `conditions` isn't huge)